### PR TITLE
compiler: catch corner case read after write

### PR DESCRIPTION
**Pull request by mloubout**
tentative, trying to see what can do for these cases

### CODE REVIEW

File: tests/test_dse.py [hunk 1]
```diff
@@ -58,6 +58,26 @@ def test_scheduling_after_rewrite():
...
+    image_vs = Function(name='image_vs', grid=grid, space_order=1, staggered=NODE)
+    p_back_xy = TimeFunction(name='p_back_xy', grid=grid, staggered=(x, y),
+                             space_order=4, time_order=1, save=Buffer(1))
+
+    eqns = [Eq(image_vs, p_back_xy + image_vs),
+            Eq(p_back_xy.backward, p_back_xy)]
+
+    op = Operator(eqns)
+
+    assert_structure(
```
Comment (EdCaunt) [inline tests/test_dse.py]: Maybe a brief comment explaining why fissioning in x is expected would be useful in the future?

File: devito/ir/support/basic.py [hunk 2]
```diff
@@ -1105,6 +1105,7 @@ def d_flow_gen(self):
                         continue
 
                     distance = dependence.distance
+
```
Comment (FabioLuporini) [inline devito/ir/support/basic.py:1112]: leftover, or apply same style change to the other sibling methods

Comment (mloubout) [reply devito/ir/support/basic.py:1112]: This actually applies the same style as `d_anti` below that has the blank line but can remove blank line from both too if prefered

File: tests/test_dse.py [hunk 3]
```diff
@@ -58,6 +58,26 @@ def test_scheduling_after_rewrite():
     assert all(trees[1].root.dim is tree.root.dim for tree in trees[1:])
 
 
+def test_scheduling_no_deriv():
```
Comment (FabioLuporini) [inline tests/test_dse.py]: maybe mention "interpolation" in the name, or it may go underestimated

**Review by FabioLuporini — APPROVED** [review]
